src/test/java/com/thoughtworks/cruise/client/TalkToCruise.java

```diff
@@ -61,10 +59,33 @@ public CruiseResponse post(String url, NameValuePair... nameValuePairs) {
             post.setRequestHeader("Accept", CruiseConstants.apiV1);
             post.setRequestHeader("Content-Type", "application/json");
         }
+
+        if(supportApiV2(url)){
+            post.setRequestHeader("Accept", CruiseConstants.apiV2);
+            post.setRequestHeader("Content-Type", "application/json");
+        }
         post.setRequestHeader("CONFIRM","true");
         return execute(url, client, post);
     }
 
+    public CruiseResponse post(String url, StringRequestEntity body) {
+        HttpClient client = client();
+        PostMethod post = new PostMethod(url);
+        post.setRequestEntity(body);
+        if(supportApiV1(url)){
```

Comment:
This seems a bit odd — I'd think we should pass in a version in the http calls from the outside in. Having to sniff URLs to figure out what api version to use is pretty odd.

Or, simply write a crud wrapper for a User:

```java
enum ApiVersion {
// see https://github.com/gocd/gocd/blob/bdf025d53142c1f233255f203fb5ec4536ab3513/api/api-base/src/main/java/com/thoughtworks/go/api/ApiVersion.java
}

abstract class Entity<T> {
  protected String id;
  protected ApiVersion apiVersion;

  protected Entity(String id, ApiVersion apiVersion) {
    this.id = id;
    this.apiVersion = apiVersion;
  }

  // allow sub classes to generate a url based on `id`
  protected abstract String url();

  // the json representation of this object
  protected abstract String toJson();
}

class User extends Entity {
  private String email;
  ...;

  public static User find(String loginName) {
    String json = talkToCruise.get(url(loginName), ApiVersion.v1);
    return fromJson(json);
  }

  public static update() {
    talkToCruise.post(this);
  }
}
```

Reply:
The side-effect of this will be that we'll have a pretty good java api that we can probably publish out as a separate project if needed.